supervisor/process.py

```diff
@@ -194,6 +195,33 @@ def spawn(self):
 
         Return the process id.  If the fork() call fails, return None.
         """
+        # this section is ownly triggered when using autostart - 
+        # otherwise the dependencies are handled in the rpcinterface
+        # check if the process is dependent upon any other process and if so,
+        # make sure that one is in the RUNNING state
+        if self.config.depends_on is not None:
+            # keep track of RUNNING childs
+            running_childs = set()
+            # wait/loop until all childs are running
+            while set(self.config.depends_on.keys()) != running_childs:
```

Comment:
The proposal looks good but code execution needs to be asynchronous. This code will stop the main supervisor loop.

Reply:
Thanks @alexsilva for the review. Stopping the main supervisor loop is not a good idea. I changed the implementation to use the rpcinterface also for spawning dependee processes when using autostart=true.  However, there might be better ways to use asyncronous calls, if you have any ideas on how to approve this any help is appreciated. Additionally there was a bug that spawn errors upon any dependee process were not properly detected. This should be fixed now.

Reply:
Quick update from myside. I was able to implement the depends_on flag without stopping the main supervisor loop. Currently testing it and then I will push the changes.